core/domain/user_services.py

```diff
@@ -98,19 +166,59 @@ def truncated_email(self):
 
     @property
     def is_known_user(self):
-        # If this does not always return True, something has gone wrong.
+        """Returns bool based on whether or not UserSettings domain
+        object contains an email property.
+
+        Returns:
+            bool. Whether this domain object contains an 'email' property.
+            If the return value is not True, something has gone wrong.
+        """
+
         return bool(self.email)
 
     @property
     def normalized_username(self):
+        """Returns username in lowercase or None if it does not exist.
+
+        Returns:
+            str or None. If this object has a 'username' property, returns 
+            the normalized version of the username. Otherwise, returns None.
+        """
+
         return self.normalize_username(self.username)
 
     @classmethod
     def normalize_username(cls, username):
+        """Returns the normalized version of the given username, 
+        or None if the passed-in 'username' is None.
+
+        Args:
+            username: str. Identifiable username to display in the UI.
+
+        Returns:
+            str or None. A string representing normalized username if it exists in this
```

Comment:
This is not what this does. The implementation makes no reference to a UserSettings domain object.